tests/failpoints/cases/test_disk_full.rs

```diff
@@ -81,7 +94,29 @@ fn test_disk_full_allowed_leader_behaviors() {
     cluster.pd_client.must_add_peer(1, new_peer(3, 3));
     must_get_equal(&cluster.get_engine(3), b"k1", b"v1");
 
-    fail::remove("disk_full_peer_1");
+    // Test operations with special flag is allowed.
```

Comment:
It's better to add a new case `test_transactions_on_full_leader` to test changes about transaction layer.